### PR TITLE
feat: subscriptions single page

### PR DESCRIPTION
**Pull request by helenosheaa**
Helps with https://github.com/influxdata/data-acquisition/issues/240

This adds a single page view for subscriptions/details view behind the feature flag: subscriptionsSinglePage which is default to false. This allows us to get feedback on two different designs for edit/details view.

Video:

https://user-images.githubusercontent.com/38860767/163249315-9084ed62-da69-4139-9d9c-b549413b21f1.mov

### CODE REVIEW

File: src/writeData/subscriptions/components/BrokerDetails.tsx [hunk 1]
```diff
@@ -66,44 +157,60 @@ const BrokerDetails: FC<Props> = ({
...
-              color={ComponentColor.Success}
-              onClick={() => {
-                history.push(`/orgs/${org.id}/notebooks`)
-              }}
-              type={ButtonType.Button}
-              testID="update-broker-form--view-data"
-              status={ComponentStatus.Default}
-            />
-          </Overlay.Footer>
+          {!singlePage ? (
```
Comment (drdelambre) [inline src/writeData/subscriptions/components/BrokerDetails.tsx:160]: at this point would it be better to have two separate components?

File: src/writeData/subscriptions/components/JsonParsingForm.tsx [hunk 2]
```diff
@@ -141,7 +143,7 @@ const JsonParsingForm: FC<Props> = ({formContent, updateForm}) => {
                   )
                 }
                 status={status}
-                maxLength={16}
+                maxLength={56}
```
Comment (genehynson) [inline src/writeData/subscriptions/components/JsonParsingForm.tsx:146]: just curious, why was 56 chars chosen?

**Review by genehynson — COMMENTED** [review]
Good stuff. I agree with Boatie - I think there might be a couple places where it would make sense to isolate the single page / non-single page stuff into their own components to help with readability. I don't expect both UI's (single page / multi page) to stay in the product for long tho so it doesn't need to be perfect

File: src/writeData/subscriptions/components/SubscriptionDetails.tsx [hunk 3]
```diff
@@ -70,56 +81,61 @@ const SubscriptionDetails: FC<Props> = ({
...
+          {!singlePage ? (
+            <Overlay.Footer>
+              <Button
+                text="Close"
+                color={ComponentColor.Tertiary}
+                onClick={() => {
+                  event('close button clicked', {}, {feature: 'subscriptions'})
+                  history.push(`/orgs/${org.id}/${LOAD_DATA}/${SUBSCRIPTIONS}`)
+                }}
+                titleText="Cancel update and return to Subscriptions list"
```
Comment (genehynson) [inline src/writeData/subscriptions/components/SubscriptionDetails.tsx:94]: I think this footer could be put into a component, maybe? It looks like the code is duplicated in each of the pages right now and is wrapped with the `{!singlePage ? ...}` flag

File: src/writeData/subscriptions/components/JsonParsingForm.tsx (hunk 2 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/components/JsonParsingForm.tsx:146]: this was already in the string parsing one so i was just making it longer based on what gary said this morning but open to suggestions

File: src/writeData/subscriptions/components/SubscriptionDetails.tsx (hunk 3 above)
Comment (helenosheaa) [reply src/writeData/subscriptions/components/SubscriptionDetails.tsx:94]: yeah i could make it a single component where the parsing info for what to navigate to next is handed in

Comment (helenosheaa) [reply src/writeData/subscriptions/components/SubscriptionDetails.tsx]: ok i've pulled it out, ended up being more faffy as the parsing form has different buttons as well.

**Comment by helenosheaa** [comment]
ok cool, thanks for the suggestions. I split out the broker one and then pulled out the footer buttons into a single component. I ended up having to change the tests as it's one and the same component for the buttons.

File: src/writeData/subscriptions/components/JsonParsingForm.tsx (hunk 2 above)
Comment (genehynson) [reply src/writeData/subscriptions/components/JsonParsingForm.tsx:146]: Looking at the db schema, the Json ones can be unlimited in length. The string ones have a max length of 255 chars. So I think we should probably mirror that here too?

**Review by genehynson — APPROVED** [review]
Love it! It's so much cleaner now - thanks :)

**Review by genehynson — APPROVED** [review]

**Pull request merged**